mesonbuild/compilers/compilers.py

```diff
@@ -234,6 +234,7 @@ def init_option(self, name: OptionKey) -> options._U:
     OptionKey('b_thinlto_cache_dir'): BaseOption(options.UserStringOption, 'Directory to store ThinLTO cache objects', ''),
     OptionKey('b_sanitize'): BaseOption(options.UserComboOption, 'Code sanitizer to use', 'none',
                                         choices=['none', 'address', 'thread', 'undefined', 'memory', 'leak', 'address,undefined']),
+    OptionKey('b_legal_code'): BaseOption(options.UserBooleanOption, 'Whether to ban dangerous constructs in a language', True),
```

Comment:
These people are going to find their code fails to compile as soon as they upgrade to 2024 compilers. It's a "don't shoot the messenger" scenario.

The constructs aren't "dangerous", they are "the C standard as released in c99, says that this is not real C code and a C compiler shouldn't successfully compile it". It's also Undefined Behavior, and a particular kind of UB that results in programs having a high tendency to crash on non-amd64 architectures. That's why GCC and clang have started actually enforcing the text of the standard, and making the compiler error out on it.

It's a user service to make that visible to people by default. There's no point in having the option if no one uses it. But an option is necessary so that people can *disable* it, if they have a codebase originally written in 1991 and cannot update it to conform to the 1999 standard.